src/thor/costmatrix.cc

```diff
@@ -432,13 +432,14 @@ void CostMatrix::CheckForwardConnections(const uint32_t source,
 
       // Special case - common edge for source and target are both initial edges
       if (pred.predecessor() == kInvalidLabel && predidx == kInvalidLabel) {
-        float s = std::abs(pred.cost().secs + opp_el.cost().secs - opp_el.transition_cost());
+        // TODO: shouldnt this use seconds? why is this using cost!?
+        float s = std::abs(pred.cost().secs + opp_el.cost().secs - opp_el.transition_cost().cost);
 
         // Update best connection and set found = true.
         // distance computation only works with the casts.
         uint32_t d = std::abs(static_cast<int>(pred.path_distance()) +
                               static_cast<int>(opp_el.path_distance()) -
-                              static_cast<int>(opp_el.transition_secs()));
+                              static_cast<int>(opp_el.transition_cost().secs));
```

Comment:
this also looks like its using the wrong thing but we'll save it for another day